lib/app/features/feed/stories/views/components/story_capture/camera/idle_camera_preview.dart

```diff
@@ -57,10 +62,38 @@ class IdleCameraPreview extends ConsumerWidget {
                 final mediaFiles =
                     await MediaPickerRoute(maxSelection: 1).push<List<MediaFile>>(context);
                 if (mediaFiles != null && mediaFiles.isNotEmpty && context.mounted) {
-                  await StoryPreviewRoute(
-                    path: mediaFiles.first.path,
-                    mimeType: mediaFiles.first.mimeType,
-                  ).push<void>(context);
+                  final filePath =
```

Comment:
Could you please move the code from 65 to 81 lines into the service class or provider to return a result ready to use on the UI side?